src/pages/registration/contributors_tab/Authors.tsx

```diff
@@ -0,0 +1,130 @@
+import deepmerge from 'deepmerge';
+import { FieldArrayRenderProps, FormikProps, useFormikContext, move } from 'formik';
+import React, { FC, useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setNotification } from '../../../redux/actions/notificationActions';
+import { Authority } from '../../../types/authority.types';
+import { Contributor, emptyContributor, UnverifiedContributor } from '../../../types/contributor.types';
+import { NotificationVariant } from '../../../types/notification.types';
+import { BackendTypeNames } from '../../../types/publication_types/commonRegistration.types';
+import { Registration } from '../../../types/registration.types';
+import { overwriteArrayMerge } from '../../../utils/formik-helpers';
+import AuthorTable from './components/AuthorTable';
+import AddIcon from '@material-ui/icons/Add';
+import styled from 'styled-components';
+import { Button } from '@material-ui/core';
+import AddContributorModal from './AddContributorModal';
+
+const StyledAddAuthorButton = styled(Button)`
+  margin-top: 1rem;
+`;
+
+interface AuthorsProps extends Pick<FieldArrayRenderProps, 'push' | 'replace'> {}
+
+const Authors: FC<AuthorsProps> = ({ push, replace }) => {
+  const { t } = useTranslation('registration');
+  const dispatch = useDispatch();
+  const { values, setValues }: FormikProps<Registration> = useFormikContext();
+  const {
+    entityDescription: { contributors },
+  } = values;
+  const [openContributorModal, setOpenContributorModal] = useState(false);
+  const [unverifiedAuthor, setUnverifiedAuthor] = useState<UnverifiedContributor | null>(null);
+  const orderedAuthors = [...contributors].map((contributor, index) => ({ ...contributor, sequence: index + 1 }));
+
+  const toggleContributorModal = () => {
+    if (unverifiedAuthor) {
+      setUnverifiedAuthor(null);
+    }
+    setOpenContributorModal(!openContributorModal);
+  };
+
+  useEffect(() => {
+    if (unverifiedAuthor) {
+      // Open modal if user has selected an unverified author
+      setOpenContributorModal(true);
```

Comment:
trenger vi egentlig `openContributorModal`? ser ut som den bare baseres på `unverifiedAuthor`, kunne vi erstatta `openContributorModal` med `!!unverifiedAuthor`?

Reply:
har ikke gjort noe med denne koden egentlig. men jo - vi kan nok simplifisere det ja!

Reply:
`setUnverifiedAuthor` blir kalt fra `AuthorCard`, så må enten ha en useEffect som lytter på den staten, eller å lage en action-funksjon som kalles og propageres videre. Foretrekker useEffect av de to egentlig.

Reply:
ja, ser det var mye gammel moro her :D men bedre å rydde opp nå enn aldri (hvis det faktisk er mulig å rydde opp)

Reply:
jo, det er sant. når jeg tenker meg om så er det vel rart at dette gjøres "på toppen" og ikke inni AuthorCard 🤔  tar en runde på det jeg